tests/topologyDescription/topologyDescription-debug-001.phpt

```diff
@@ -0,0 +1,38 @@
+--TEST--
+MongoDB\Driver\TopologyDescription debug output
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$manager = create_test_manager();
+
+class MySubscriber implements MongoDB\Driver\Monitoring\SDAMSubscriber
+{
+    private $topologyDescription;
+
+    public function topologyChanged(MongoDB\Driver\Monitoring\TopologyChangedEvent $event)
+    {
+        if (! $this->topologyDescription) {
+            $this->topologyDescription = $event->getNewDescription();
+            var_dump($this->topologyDescription);
+        }
+    }
+}
+
+$subscriber = new MySubscriber;
+$manager->addSubscriber($subscriber);
+
+$command = new MongoDB\Driver\Command(['ping' => 1]);
+$manager->executeCommand(DATABASE_NAME, $command);
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECTF--
+object(MongoDB\Driver\TopologyDescription)#%d (%d) {
+  %a
```

Comment:
In the `var_export` test below you asserted that the debug output at least contained "servers" and "type" fields. Why is this empty?

The `get_debug_info` and `get_properties` handlers both call `php_phongo_topologydescription_get_properties_hash`, so there isn't much technical difference between `var_dump` and `var_export` apart from how PHP will present returned HashTable. But using `%a` here essentially isn't testing `php_phongo_topologydescription_get_properties_hash` at all, since we're not expecting _any_ fields.

This relates my other other comment about `%a` and `%A` being very liberal wildcards and generally not something to use this high up. If you look at other uses of them within the test suite I think you'll generally find them used for matching arbitrary binary data (which could include newline characters) or the body of `Server::getInfo()` (which originates from the server).

If you end up deleting the `var_export` test as I suggest, this should be revised in turn to expect those fields. Also, if you change the test to capture the final TopologyDescription instead of the first, you should be able to assert at least one element in the server object (since that'd be common to any topology we connect to and run a `ping` command). And the `type` field could use `%r` to expect one of the three possible strings we'd expect as well.

Reply:
Sorry, I must have missed updating this test to be similar to the `var_export` one, but was thinking that all the tests would probably need to be improved so I would come back to it eventually.

Reply:
Noted. In this case it would have been helpful to preemptively open a comment thread to communicate your intention (especially since a "todo" comment can't be left here). Allowing the comment to stay unresolved until it's addressed would serve double duty to inform reviewers and also serve as a personal reminder.